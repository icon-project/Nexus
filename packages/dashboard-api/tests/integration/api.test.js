'use strict';

const request = require('supertest');
const HttpStatus = require('@tiendq/http-status');
const app = require('../../src/app/app');

/***************************************** Test btpnetwork *********************************/
//function add Method expect any for array minted
function addMethodAnyForMint(arr) {
  for (let i = 0; i < arr.length; i++) {
    expect(arr[i]).toMatchSnapshot({
      mintedVolume: expect.any(Number),
      networkId: expect.any(String),
      networkName: expect.any(String),
    });
  }
}

describe('Test btpNetwork', () => {
  test('GET /btpnetwork', async () => {
    let response = await request(app)
      .get('/v1/btpnetwork')
      .set('Accept', 'application/json')
      .expect(HttpStatus.OK);
    //Using toMatchSnapshot to save Snapshots in folder _snapshots_ and then compare for next time
    expect(response.body.content).toMatchSnapshot({
      bondedValue: expect.any(Number),
      fee: expect.any(Object),
      minted: expect.any(Array),
      totalNetworks: expect.any(Number),
      totalTransactions: expect.any(Number),
      volume: expect.any(Number),
    });

    //If snapshots have array, We will use toMatchSnapshot for each element of the array
    const objectMintedInArray = response.body.content.minted;
    addMethodAnyForMint(objectMintedInArray);
  });

  test('GET /btpnetwork?stats=1', async () => {
    let response = await request(app)
      .get('/v1/btpnetwork')
      .query({
        stats: 1,
      })
      .set('Accept', 'application/json')
      .expect(HttpStatus.OK);

    expect(response.body.content).toMatchSnapshot({
      bondedValue: expect.any(Number),
      fee: {
        allTimeAmount: expect.any(Array),
        assets: expect.any(Array),
        cumulativeAmount: expect.any(Number),
        currentAmount: expect.any(Number),
      },
      minted: expect.any(Array),
      stats: expect.any(Object),
      totalNetworks: expect.any(Number),
      totalTransactions: expect.any(Number),
      volume: expect.any(Number),
    });

    const objectMintedInArray = response.body.content.minted;

    addMethodAnyForMint(objectMintedInArray);

    const objectIndexersInArray = response.body.content.stats.indexers;
    for (let i = 0; i < objectIndexersInArray.length; i++) {
      expect(objectIndexersInArray[i]).toMatchSnapshot({
        block_height: expect.any(Number),
        name: expect.any(String),
        network_id: expect.any(String),
        updated_time: expect.any(String),
      });
    }
  });

  test('GET /btpnetwork?volumeLast24h=true&mintLast24h=true', async () => {
    let response = await request(app)
      .get('/v1/btpnetwork')
      .query({
        volumeLast24h: true,
        mintLast24h: true,
      })
      .set('Accept', 'application/json')
      .expect(HttpStatus.OK);

    expect(response.body.content).toMatchSnapshot({
      bondedValue: expect.any(Number),
      fee: {
        allTimeAmount: expect.any(Array),
        assets: expect.any(Array),
        cumulativeAmount: expect.any(Number),
        currentAmount: expect.any(Number),
      },
      mintVolumeLast24hChange: expect.any(Number),
      minted: expect.any(Array),
      totalNetworks: expect.any(Number),
      totalTransactions: expect.any(Number),
      volume: expect.any(Number),
      volumeLast24hChange: expect.any(Number),
    });

    const objectMintedInArray = response.body.content.minted;

    addMethodAnyForMint(objectMintedInArray);
  });
  test('GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd', async () => {
    let response = await request(app)
      .get('/v1/btpnetwork')
      .query({
        token: 'btc',
        amount: 100,
        convert_to: 'usd',
      })
      .set('Accept', 'application/json')
      .expect(HttpStatus.OK);

    expect(response.body.content).toMatchSnapshot({
      bondedValue: expect.any(Number),
      fee: {
        allTimeAmount: expect.any(Array),
        assets: expect.any(Array),
        cumulativeAmount: expect.any(Number),
        currentAmount: expect.any(Number),
      },
      minted: expect.any(Array),
      totalNetworks: expect.any(Number),
      totalTransactions: expect.any(Number),
      volume: expect.any(Number),
    });

    const objectMintedInArray = response.body.content.minted;
    addMethodAnyForMint(objectMintedInArray);
  });
});
