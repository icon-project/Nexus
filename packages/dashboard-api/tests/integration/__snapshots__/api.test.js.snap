// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test btpNetwork GET /btpnetwork 1`] = `
Object {
  "bondedValue": Any<Number>,
  "fee": Any<Object>,
  "minted": Any<Array>,
  "totalNetworks": Any<Number>,
  "totalTransactions": Any<Number>,
  "volume": Any<Number>,
}
`;

exports[`Test btpNetwork GET /btpnetwork 2`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork 3`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork 4`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork 5`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd  1`] = `
Object {
  "bondedValue": Any<Number>,
  "fee": Object {
    "allTimeAmount": Any<Array>,
    "assets": Any<Array>,
    "cumulativeAmount": Any<Number>,
    "currentAmount": Any<Number>,
  },
  "minted": Any<Array>,
  "totalNetworks": Any<Number>,
  "totalTransactions": Any<Number>,
  "volume": Any<Number>,
}
`;

exports[`Test btpNetwork GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd  2`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd  3`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd  4`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork/converter?token=btc&amount=100&convert_to=usd  5`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 1`] = `
Object {
  "bondedValue": Any<Number>,
  "fee": Object {
    "allTimeAmount": Any<Array>,
    "assets": Any<Array>,
    "cumulativeAmount": Any<Number>,
    "currentAmount": Any<Number>,
  },
  "minted": Any<Array>,
  "stats": Any<Object>,
  "totalNetworks": Any<Number>,
  "totalTransactions": Any<Number>,
  "volume": Any<Number>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 2`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 3`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 4`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 5`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 6`] = `
Object {
  "block_height": Any<Number>,
  "name": Any<String>,
  "network_id": Any<String>,
  "updated_time": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 7`] = `
Object {
  "block_height": Any<Number>,
  "name": Any<String>,
  "network_id": Any<String>,
  "updated_time": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?stats=1 8`] = `
Object {
  "block_height": Any<Number>,
  "name": Any<String>,
  "network_id": Any<String>,
  "updated_time": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?volumeLast24h=true&mintLast24h=true 1`] = `
Object {
  "bondedValue": Any<Number>,
  "fee": Object {
    "allTimeAmount": Any<Array>,
    "assets": Any<Array>,
    "cumulativeAmount": Any<Number>,
    "currentAmount": Any<Number>,
  },
  "mintVolumeLast24hChange": Any<Number>,
  "minted": Any<Array>,
  "totalNetworks": Any<Number>,
  "totalTransactions": Any<Number>,
  "volume": Any<Number>,
  "volumeLast24hChange": Any<Number>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?volumeLast24h=true&mintLast24h=true 2`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?volumeLast24h=true&mintLast24h=true 3`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?volumeLast24h=true&mintLast24h=true 4`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;

exports[`Test btpNetwork GET /btpnetwork?volumeLast24h=true&mintLast24h=true 5`] = `
Object {
  "mintedVolume": Any<Number>,
  "networkId": Any<String>,
  "networkName": Any<String>,
}
`;
